include/nbl/video/alloc/StreamingTransientDataBuffer.h

```diff
@@ -97,7 +97,7 @@ class StreamingTransientDataBufferST : protected SubAllocatedDataBuffer<core::He
 
 
 template< typename _size_type=uint32_t, class CPUAllocator=core::allocator<uint8_t>, class CustomDeferredFreeFunctor=void, class RecursiveLockable=std::recursive_mutex>
-class StreamingTransientDataBufferMT : protected StreamingTransientDataBufferST<_size_type,CPUAllocator,CustomDeferredFreeFunctor>, public virtual core::IReferenceCounted
+class StreamingTransientDataBufferMT : public StreamingTransientDataBufferST<_size_type,CPUAllocator,CustomDeferredFreeFunctor>, public virtual core::IReferenceCounted
```

Comment:
protected inheritance for a reason

Reply:
the reason for changing protected to public was because a constructor from the protected class was used externally. Clang marked it as a compile time error

Reply:
ah yes, the `using Base::Base;` is a problem when the base is not public (when you do `using Base::method` you inherit the private/protected/public scoping)

In that case you'll need to write a perfect forwarding constructor
```cpp
template<typename... Args>
StreamingTransientDataBufferMT::StreamingTransientDataBufferMT(Args&&... args) : Base(std::forward<Args>(args)...) {}
```

and keep the Base protected as before

Reply:
oh, i was wrong, it was not the constructor being inaccessible, it was the `operator new` of `nbl::core::impl::ResolveAlignment<nbl::core::AlignedBase<16> >` which was inherited from protected base class. The new operator is being used in `make_smart_refctd_ptr`

Reply:
solved with the passhtrough